### PR TITLE
fix(VIPERGC-574): update fault provider for new YAMCS API and provide shelve durations

### PR DESCRIPTION
**Pull request by ozyx**
Closes VIPERGC-574 

### Describe your changes:

- Updates the openmct-yamcs fault provider to match the API changes from YAMCS, fixing acknowledges and indefinite shelving of faults.
- Adds e2e tests that use and mock real yamcs network traffic to test the various fault actions
- Adds a `getShelveDuration()` method to allow `openmct-yamcs` fault provider to specify its own shelve durations.

**NOTE**: https://github.com/nasa/openmct/pull/7849 must be merged first

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this a [notable change](../docs/src/process/release.md) that will require a special callout in the release notes? For example, will this break compatibility with existing APIs or projects that consume these plugins?

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?

### CODE REVIEW

**Comment by ozyx** [comment]
https://github.com/nasa/openmct/pull/7849 needs to be merged before tests pass here

File: src/openmct-yamcs.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ export default function install(
 
         openmct.faults.addProvider(new YamcsFaultProvider(openmct,
             {
-                faultModelConvertor,
+                convertDataToFaultModel,
```
Comment (shefalijoshi) [inline src/openmct-yamcs.js:79]: Looks like a mismatch of parameters provided to the exoectation here: https://github.com/akhenry/openmct-yamcs/pull/481/files#diff-e3379e1f900a4b0e58c3057f59da360a685831b58d75bf3b80d0bfdaeda4ba88R6

**Review by shefalijoshi — COMMENTED** [review]
A few comments. Nice work!

File: src/providers/fault-mgmt-providers/realtime-fault-provider.js [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
 import { DATA_TYPES, NAMESPACE, OBJECT_TYPES } from '../../const.js';
+import { convertDataToFaultModel } from './utils.js';
 
 export default class RealtimeFaultProvider {
     #openmct;
-    constructor(openmct, faultModelConverter, instance) {
+    constructor(openmct, instance) {
         this.#openmct = openmct;
-        this.faultModelConverter = faultModelConverter;
+        this.convertDataToFaultModel = convertDataToFaultModel;
```
[the remark this answers is not in the document]
Comment (ozyx) [reply src/providers/fault-mgmt-providers/realtime-fault-provider.js:9]: Ah, good catch

File: src/openmct-yamcs.js (hunk 1 above)
Comment (ozyx) [reply src/openmct-yamcs.js:79]: wow, really good catch

File: src/providers/fault-mgmt-providers/fault-action-provider.js [hunk 3]
```diff
@@ -9,52 +9,92 @@ export default class FaultActionProvider {
...
-
-        return url;
+    /**
+     * @param {FaultModel} fault the fault to perform the action on
+     * @param {'acknowledge' | 'shelve' | 'unshelve' | 'clear'} action the action to perform on the fault
+     * @returns {string} the URL to perform the action on the fault
+     */
+    #getUrl(fault, action) {
+        return `${this.url}api/processors/${this.instance}/${this.processor}/${FAULT_MGMT_ALARMS}`
+               + `${fault.namespace}/${fault.name}/${fault.seqNum}:${action}`;
```
[the remark this answers is not in the document]
Comment (ozyx) [reply src/providers/fault-mgmt-providers/fault-action-provider.js:92]: kind of confusing docs from YAMCS API, but I believe this always starts with a `/`:

```
  // For parameter alarms, this is the id of the parameters
  // For event alarms
  //   - the id.namespace is /yamcs/event/<EVENT_SOURCE>, unless
  //     EVENT_SOURCE starts with a "/" in which case the namespace
  //     is just the <EVENT_SOURCE>
  //   - the id.name is the <EVENT_TYPE>
  id: NamedObjectId;
  ```

**Review by shefalijoshi — APPROVED** [review]
Looks good, thanks.

**Pull request merged**